rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -569,7 +569,7 @@ def parse_primitive_value_string(type_, value_string):
                 tmp_value_string = value_string
                 clear_escaping = False
                 index = 0
-                while index > -1:
+                while index != -1:
```

Comment:
From reading the code I think this condition doesn't make any difference. It could be changed to `while True`. It is only relevant when the loop enters the first time. After that the `index` is always changed in the next line and then tested in the over next. After that `index` is not changed anymore so if the code doesn't hit the `break` two lines below it will always do another loop.